### PR TITLE
add in MCif Delete ,authorization dan Mapping builder

### PR DESCRIPTION
**Pull request by Yaksyin**
add in TSavingAccount Put/Edit update
add MCif Delete , authorization dan  maping build
image_2024-12-14_051901540
image
image

{
    "accountNumber": "4444567890",
    "cifId": "1820741",
    "savingId": "431213"
}
RESPONSE :
{
    "savingAccountId": "97149824619284",
    "accountNumber": "4444567890",
    "cifId": "1820741",
    "beginBalance": 0.00,
    "endBalance": 0.00,
    "currentBalance": 0.00,
    "statusId": "1",
    "statusName": "Pending",
    "authorizationAt": null,
    "authorizationBy": "",
    "savingId": "expresi",
    "savingName": "expresi",
    "deleted": false
}

### CODE REVIEW

File: src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 public class TSavingAccountController {
     private TSavingAccountService tSavingAccountService;
 
-    TSavingAccountController(TSavingAccountService tSavingAccountService) {
+   TSavingAccountController(TSavingAccountService tSavingAccountService) {
         this.tSavingAccountService = tSavingAccountService;
     }
+
+    @PutMapping("/{savingAccountId}")
+    public TSavingAccountResponse createTSavingAccount(@PathVariable String savingAccountId,
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java]: so is this controller for create or edit?

**Review by Jundillah123 — APPROVED** [review]

**Review by 261001 — APPROVED** [review]

File: src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package com.b2camp.simple_core_banking.controller;
 
 
+import com.b2camp.simple_core_banking.dto.MCifResponse;
 import com.b2camp.simple_core_banking.service.MCifService;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java]: create single import

File: src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java [hunk 3]
```diff
@@ -23,10 +23,7 @@ public class MCifResponse {
     private String statusId;
     private String statusName;
 
-    public String getCifId() {
-
-        return cifId;
-    }
+    public String getCifId() { return cifId; }
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java]: kenapa ini masih pake getter setter manual? bukannya di develop sudah pake lombok?

File: src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountResponse.java [hunk 4]
```diff
@@ -19,6 +20,15 @@ public class TSavingAccountResponse {
 
     //
 
+
+    public String getStatusName() {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountResponse.java]: kenapa ini masih pake getter setter manual? bukannya di develop sudah pake lombok?

File: src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java [hunk 5]
```diff
@@ -1,4 +1,26 @@
...
+    String deleted(String savingAccountId);
+
+    TSavingAccountResponse authorization(String savingAccountId);
+    TSavingAccountResponse create(TSavingAccountRequest tSavingAccountRequest);
+
+    List<TSavingAccountResponse> readTSavingAccount(String accountNumber);
+
+    Optional<TSavingAccountResponse> findBySavingAccountId(String savingAccountId);
+
+    @Transactional(propagation = Propagation.REQUIRED,rollbackFor = Exception.class)
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java]: ini kenapa? @Transactional di simpen di service?

File: src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java [hunk 6]
```diff
@@ -1,6 +1,255 @@
...
+        }
+
+        return tSavingAccountResponse;
+    }
+
+    @Override
+    @Transactional (propagation = Propagation.REQUIRED,rollbackFor = Exception.class)
+    public TSavingAccountResponse updateSavingAccount(String savingAccountId, TSavingAccountRequest request) {
+        TSavingAccount tSavingAccount = tSavingAccountRepository
+                .findById(savingAccountId).orElse(null);
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java]: ini udah mencari data berdasarkan isDeleted is false?

File: untitled/pom.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (xuibdg) [inline untitled/pom.xml]: kenapa menambahkan untitled/pom.xml?

**Pull request closed without merging**